partcad-cad-freecad/partcad_freecad/provision.py

```
@@ -0,0 +1,446 @@
+            # entries with whatever the umask gave them.
+            mode = (info.external_attr >> 16) & 0o777
+            if mode:
+                os.chmod(path, mode)
+
+
+def _extract_tar(archive: str, dest: str) -> None:
+    with tarfile.open(archive, "r:*") as tf:
+        try:
+            tf.extractall(dest, filter="data")  # nosec B202 - trusted, checksum-verified archive
+        except TypeError:  # pragma: no cover - Python < 3.12 has no filter argument
+            tf.extractall(dest)  # nosec B202 - trusted, checksum-verified archive
```

Ruff (0.16.1)

[error] Uses of `tarfile.extractall()`

(S202)